### PR TITLE
Implement Cloud Trails

### PR DESCRIPTION
**Pull request by s2504s**
## What
* Implement a TF module to enable the AWS CloudTrail service and ship logs to an S3 bucket

## Why
* Provide a way to easily enable AWS API audit logs in a consistent fashion

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,51 @@
+module "tf_label" {
+  source    = "../tf_label"
```
Comment (osterman) [inline main.tf]: Use git URI syntax

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "${var.name}"
+}
+
+resource "aws_cloudtrail" "ct" {
```
Comment (osterman) [inline main.tf]: lets just call it `default`

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  name                          = "${module.tf_label.id}"
+  s3_bucket_name                = "${aws_s3_bucket.cloud_trail.id}"
+  enable_logging                = "${var.enable_logging}"
+  enable_log_file_validation    = "${var.enable_log_file_validation}"
+  is_multi_region_trail         = "${var.is_multi_region_trail}"
+  include_global_service_events = "${var.include_global_service_events}"
+}
+
+resource "aws_s3_bucket" "cloud_trail" {
```
Comment (osterman) [inline main.tf]: Let's just call it `default` since there is only one bucket used.

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,31 @@
+variable "region" {
+  default = ""
+}
+
+variable "name" {
+  default = ""
```
Comment (osterman) [inline variables.tf]: Default the name to `cloudtrails`

File: README.md [hunk 5]
```diff
@@ -1 +1,33 @@
...
+#### Optional
+
+* enable_logging: Enable logging, set to 'false' to Pause logging (default = true)
+* enable_log_file_validation: Create signed digest file to validated contents of logs (default = true)
+* include_global_service_events: Include evnets from global services such as IAM (default = false)
+* is_multi_region_trail: Whether the trail is created in all regions or just the current region (default = false)
+
+## Usage
+```
+module "ct" {
```
Comment (osterman) [inline README.md]: call it `cloudtrails`. normalize the spacing the way that `terraform fmt` does it.

File: README.md [hunk 6]
```diff
@@ -1 +1,33 @@
...
+  source = "github.com/cloudposse/tf_cloudtrail"
```
Comment (osterman) [inline README.md]: prefer you use explicit notation. 

e.g.
```
source    = "git::https://github.com/cloudposse/tf_label.git?ref=master"
```

File: README.md [hunk 7]
```diff
@@ -1 +1,33 @@
...
+#### Required
+
+* name: Name of CloudTrail trail.
+* region: AWS region. Used to find remote state.
+
+
+
+#### Optional
+
+* enable_logging: Enable logging, set to 'false' to Pause logging (default = true)
```
Comment (osterman) [inline README.md]: Use markdown formatting for code. 
E.g. `enable_logging`

@comeanother please help format explanation.

**Comment by osterman** [comment]
- [ ] Fix `LICENSE` copyright.

File: README.md [hunk 8]
```diff
@@ -1 +1,33 @@
...
+#### Required
+
+* `name`: Name of CloudTrail trail.
+* `region`: AWS region. Used to find remote state.
+
+
+
+#### Optional
+
+* `enable_logging`: Enable logging, set to 'false' to Pause logging (default = true)
```
Comment (osterman) [inline README.md]: `false` is also a keyword, so it should be markdown too. Same with `default = true`

File: README.md [hunk 9]
```diff
@@ -1 +1,33 @@
...
+
+#### Required
+
+* `name`: Name of CloudTrail trail.
+* `region`: AWS region. Used to find remote state.
+
+
+
+#### Optional
+
```
Comment (osterman) [inline README.md]: Please "borrow" the same language used on the terraform docs. See: https://www.terraform.io/docs/providers/aws/r/cloudtrail.html

File: main.tf [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  enable_log_file_validation    = "${var.enable_log_file_validation}"
+  is_multi_region_trail         = "${var.is_multi_region_trail}"
+  include_global_service_events = "${var.include_global_service_events}"
+}
+
+resource "aws_s3_bucket" "default" {
+  bucket        = "${module.tf_label.id}"
+  force_destroy = false
+
+  policy = <<POLICY
```
Comment (osterman) [inline main.tf]: Don't use `HEREDOC` when a better resource type exists. https://www.terraform.io/docs/providers/aws/d/iam_policy_document.html

File: main.tf [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+resource "aws_cloudtrail" "default" {
+  name                          = "${module.tf_label.id}"
+  s3_bucket_name                = "${aws_s3_bucket.default.id}"
+  enable_logging                = "${var.enable_logging}"
+  enable_log_file_validation    = "${var.enable_log_file_validation}"
+  is_multi_region_trail         = "${var.is_multi_region_trail}"
+  include_global_service_events = "${var.include_global_service_events}"
+}
+
+data "aws_iam_policy_document" "cloudtrail" {
```
Comment (osterman) [inline main.tf]: call it `default` since there's only one.

File: main.tf [hunk 12]
```diff
@@ -0,0 +1,67 @@
+module "tf_label" {
+  source    = "git::https://github.com/cloudposse/tf_label.git?ref=master"
```
Comment (osterman) [inline main.tf]: Let's pin to 0.1.0

File: outputs.tf [hunk 13]
```diff
@@ -0,0 +1,3 @@
+output "s3_trail_bucket_name" {
```
Comment (osterman) [inline outputs.tf]: Let's just call it `bucket_name` since there is no intention to have more than one bucket.

File: README.md [hunk 14]
```diff
@@ -1 +1,26 @@
-# tf_cloudtrail
\ No newline at end of file
+# tf_cloudtrail
+
+Setup and manage CloudTrail
+
+### Argument Reference
```
Comment (comeanother) [inline README.md]: Use second-level headers
`##` instead of `###`

**Pull request merged**